### PR TITLE
feat(web): Teach studio to connect to Novu API for execution details

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

Teach studio to connect to Novu API for execution details.

### CODE REVIEW

File: apps/web/src/api/api.client.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ interface IOptions {
   absoluteUrl: boolean;
 }
 
+// @deprecated Migrate all api methods to the new buildAPIHTTPClient that allows runtime configuration on the client object.
```
Comment (SokratisVidros) [inline apps/web/src/api/api.client.ts:10]: This post-release clean-up work needs to happen to allow us to have way more predictable studio components that use a single source of hooks to speak to bridge or Novu API.

File: apps/web/src/api/api.client.ts [hunk 2]
```diff
@@ -91,3 +92,49 @@ function getHeaders() {
       }
     : {};
 }
+
+// WIP: The static API client needs to be replaced by a dynamic API client where api keys are injected.
+export function buildAPIHTTPClient({
```
Comment (rifont) [inline apps/web/src/api/api.client.ts:97]: 🤓 **nitpick (non-blocking):** Prefer camel-cased acronyms for better readability‏
```suggestion
export function buildApiHttpClient({
```

File: apps/web/src/studio/LocalStudioAuthenticator.tsx [hunk 3]
```diff
@@ -75,6 +77,15 @@ export function LocalStudioAuthenticator() {
...
     }
 
+    // Wait for environment and apiKeys to be loaded
+    if (!environment || !apiKey) {
+      return;
+    }
+
+    if (environment.name.toLowerCase() !== 'development') {
+      throw new Error('Local Studio works only with development api keys');
+    }
```
Comment (rifont) [inline apps/web/src/studio/LocalStudioAuthenticator.tsx:87]: 👏 **praise:** Lovely!‏

**Review by rifont — APPROVED** [review]
🚀

**Pull request merged**